In [7]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [8]:
iris = load_iris()


##### Logistic Regression

In [13]:
l_scores = cross_val_score(LogisticRegression(max_iter=7600), iris.data, iris.target,cv=3)
l_scores

array([0.98, 0.96, 0.98])

In [14]:
np.average(l_scores)

0.9733333333333333

##### Decision Tree

In [17]:
d_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target,cv=3)
d_scores

array([0.98, 0.94, 0.98])

In [18]:
np.average(d_scores)

0.9666666666666667

##### Support Vector Machine (SVM)

In [19]:
s_scores = cross_val_score(SVC(), iris.data, iris.target,cv=3)
s_scores

array([0.96, 0.98, 0.94])

In [20]:
np.average(s_scores)

0.96

##### Random Forest

In [22]:
r_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target,cv=3)
r_scores

array([0.98, 0.94, 0.96])

In [23]:
np.average(r_scores)

0.96

##### Best score so far is from Logistic Regression : 0.9733333333333333 